src/sys.c

```diff
@@ -708,28 +708,14 @@ JL_DLLEXPORT void jl_raise_debugger(void)
 #endif // _OS_WINDOWS_
 }
 
-JL_DLLEXPORT jl_sym_t *jl_get_OS_NAME(void)
+JL_DLLEXPORT jl_sym_t *jl_get_UNAME(void)
 {
-#if defined(_OS_WINDOWS_)
-    return jl_symbol("Windows");
-#elif defined(_OS_LINUX_)
-    return jl_symbol("Linux");
-#elif defined(_OS_FREEBSD_)
-    return jl_symbol("FreeBSD");
-#elif defined(_OS_DARWIN_)
-    return jl_symbol("Darwin");
-#else
-#warning OS_NAME is Unknown
-    return jl_symbol("Unknown");
-#endif
+    return jl_symbol(JL_BUILD_UNAME);
```

Comment:
You'll get a corrupted build then (for other reasons). I don't care about that case. I prefer that it breaks noisily instead.